### PR TITLE
Fixed bugs found by PVS-Studio

### PR DESCRIPTION
**Pull request by alexey-milovidov**
For changelog. Remove if this is non-significant change.

Category (leave one):
- Build/Testing/Packaging Improvement

Short description (up to few sentences):
Fixed bugs found by PVS-Studio.

### CODE REVIEW

File: dbms/src/Compression/CompressionCodecZSTD.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ CompressionCodecZSTD::CompressionCodecZSTD(int level_)
 
 void registerCodecZSTD(CompressionCodecFactory & factory)
 {
-    UInt8 method_code = static_cast<char>(CompressionMethodByte::ZSTD);
+    UInt8 method_code = UInt8(CompressionMethodByte::ZSTD);
```
Comment (abyss7) [inline dbms/src/Compression/CompressionCodecZSTD.cpp]: C-style cast doesn't look good

File: dbms/src/Interpreters/InterpreterSystemQuery.cpp [hunk 2]
```diff
@@ -289,7 +290,7 @@ void InterpreterSystemQuery::restartReplicas(Context & system_context)
     if (replica_names.empty())
         return;
 
-    ThreadPool pool(std::min(getNumberOfPhysicalCPUCores(), replica_names.size()));
+    ThreadPool pool(std::min(size_t(getNumberOfPhysicalCPUCores()), replica_names.size()));
```
Comment (abyss7) [inline dbms/src/Interpreters/InterpreterSystemQuery.cpp]: C-style cast doesn't look good - may be `std::min<size_t>(…)` is better?

File: dbms/src/Storages/Kafka/StorageKafka.cpp [hunk 3]
```diff
@@ -304,7 +304,7 @@ BlockInputStreams StorageKafka::read(
     if (num_created_consumers == 0)
         return BlockInputStreams();
 
-    const size_t stream_count = std::min(num_streams, num_created_consumers);
+    const size_t stream_count = std::min(size_t(num_streams), num_created_consumers);
```
Comment (abyss7) [inline dbms/src/Storages/Kafka/StorageKafka.cpp]: C-style cast doesn't look good - may be `std::min<size_t>(…)` is better?

File: dbms/src/Storages/StorageMerge.cpp [hunk 4]
```diff
@@ -219,7 +220,7 @@ BlockInputStreams StorageMerge::read(
         size_t current_need_streams = tables_count >= num_streams ? 1 : (num_streams / tables_count);
         size_t current_streams = std::min(current_need_streams, remaining_streams);
         remaining_streams -= current_streams;
-        current_streams = std::max(1, current_streams);
+        current_streams = std::max(size_t(1), current_streams);
```
Comment (abyss7) [inline dbms/src/Storages/StorageMerge.cpp]: C-style cast doesn't look good - may be `std::max<size_t>(…)` is better?

**Review by abyss7 — CHANGES_REQUESTED** [review]
Found some C-style casts

File: dbms/src/Compression/CompressionCodecZSTD.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply dbms/src/Compression/CompressionCodecZSTD.cpp]: This is not a C-style cast. It is named "constructor cast".

File: dbms/src/Interpreters/InterpreterSystemQuery.cpp (hunk 2 above)
Comment (alexey-milovidov) [reply dbms/src/Interpreters/InterpreterSystemQuery.cpp]: std::min takes both arguments by references.

**Pull request merged**